Automattic/wp-calypso
# Email: Redesign section headers and add links to services in Email page

This pull request updates the [`Email` page](https://wordpress.com/email/) for Titan and Google Workspace in order to:

* **Display icons of email providers**: this improves branding as well as consistency with the `Purchases Settings` page
* **Show product names and domain names on two lines**: this is to avoid issues with long names and small viewports
* **Add an `Add New Mailboxes` button for Titan**: that way the user experience is more consistent across email providers
* **Add icons and links to email provider services**: so users can access their mailboxes and other apps easily

Before | After | After (Expanded)
------ | ----- | -----
![B](https://user-images.githubusercontent.com/594356/107667485-e7cc9680-6c8f-11eb-8b2a-cc22fe19e848.png) | ![A](https://user-images.githubusercontent.com/594356/107667517-ed29e100-6c8f-11eb-854f-9d7830f2c1fe.png) | ![screenshot](https://user-images.githubusercontent.com/594356/107982256-fc789980-6fc3-11eb-9a39-8d5fbd4d94e9.png)


Note the design was inspired by the [`Connections` page](https://wordpress.com/marketing/connections/). There is some duplicate code between the G Suite and Titan components. Ideally, we should reuse the same components but since this pull request is already long enough I thought this refactoring should take place in another one.

#### Testing instructions

1. Run `git checkout update/email-page` and start your server, or open a [live branch](https://calypso.live/?branch=update/email-page)
2. Log into a WordPress.com account with a Titan or a Google Workspace account
3. Open the [`Email` page](http://calypso.localhost:3000/email)
4. Assert that this page looks like the screenshots above
5. Assert that it still looks fine on small viewports
6. Check that the placeholder displays correctly
